From 81476d67c4db9f5ec0068deda849b6578a9b9d2a Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sun, 1 Dec 2019 11:55:49 -0600
Subject: [PATCH] Workaround an ICE in clang 9.0.0

This bug is not there in 8.x nor in the 9.0 daily snapshot.
---
 kernel/x86_64/dsymv_L_microk_skylakex-2.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/kernel/x86_64/dsymv_L_microk_skylakex-2.c b/kernel/x86_64/dsymv_L_microk_skylakex-2.c
index 8244dffa1..8f205e28e 100644
--- a/kernel/x86_64/dsymv_L_microk_skylakex-2.c
+++ b/kernel/x86_64/dsymv_L_microk_skylakex-2.c
@@ -27,7 +27,8 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 /* need a new enough GCC for avx512 support */
-#if (( defined(__GNUC__)  && __GNUC__   > 6 && defined(__AVX2__)) || (defined(__clang__) && __clang_major__ >= 6))
+#if (( defined(__GNUC__)  && __GNUC__   > 6 && defined(__AVX2__)) || (defined(__clang__) && \
+ (__clang_major__ >= 6 || !(__clang_major__ == 9 && __clang_minor__ == 0 && __clang_patchlevel__ == 0))))
 
 #include <immintrin.h>
 
@@ -158,4 +159,4 @@ static void dsymv_kernel_4x4(BLASLONG from, BLASLONG to, FLOAT **a, FLOAT *x, FL
 } 
 #else
 #include "dsymv_L_microk_haswell-2.c"
-#endif
\ No newline at end of file
+#endif
